frontends/web/src/components/forms/checkbox.test.tsx

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright 2022 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { render, screen } from '@testing-library/react';
+import Checkbox from './checkbox';
+
+describe('components/forms/checkbox', () => {
+  it('renders checkbox with proper attributes', () => {
+    const { container } = render(<Checkbox label="my checkbox label" id="checkbox" />);
+    expect(container.firstChild).toHaveClass('default');
+    expect(screen.getByRole('checkbox', { name: 'my checkbox label' })).toBeInTheDocument();
```

Comment:
Sorry I don't know testing lib well, can you explain what exactly is tested to be inthedocument?

I don't understand exactly what getByRole finds, but I am thinking about DOM elements, maybe that is wrong?

The component renders a span containing `input[type="checkbox"]` element with a sibling  `label` element that contains the text.  So does getByRole find the "input type checkbox", the label element or the whole component or something in the "accessibility tree" (found https://testing-library.com/docs/queries/about#priority 🤔  )

Reply:
No worries, I am also still learning 🙂 

`toBeInTheDocument` is quite simply asserting that whatever it is (in this case checkbox with a label "my checkbox label") to be within the document body, aka in the page. 

I haven't tested but I think here are all the "valid roles" in HTML5: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles

>> So does getByRole find the "input type checkbox", the label element or the whole component or something in the "accessibility tree" (found...

That link you attached described exactly what it is and why it's the preferred method in most situations.

In this case, byRole checks the whole screen and see if there's a **checkbox** with the label **my checkbox label**. byRole can "connect" these two (label and checkbox) because of the `htmlFor` we use in the `label` and `id` in the `input checkbox` (`htmlFor` === `id`). 

byRole doesn't "see" per element (as opposed to getByTestId, getByid and similars), but it tries to act like a "user" e.g looking at it holistically. In this case, it's looking for "a checkbox with a label of bla bla..". Here, we can tell that we can only use this way of testing (byRole) if we follow accessibility best practices, which we should always try to do anyway.